STYLE_GUIDE.md

```diff
@@ -0,0 +1,173 @@
+- If you want to explain the commit in more depth, following the first line should be a blank line and then a more detailed description of the commit. This can be as detailed as you want, so dig into details here and keep the first line short.
+
+### Rebase upstream changes into your branch
+
+Once you are done making changes, you can begin the process of getting
+your code merged into the main repo. Step 1 is to rebase upstream
+changes to the master branch into yours by running this command
+from your branch:
+
+```bash
+git pull --rebase upstream master
+```
+
+This will start the rebase process. You must commit all of your changes
+before doing this. If there are no conflicts, this should just roll all
+of your changes back on top of the changes from upstream, leading to a
+nice, clean, linear commit history.
+
+If there are conflicting changes, git will start yelling at you part way
+through the rebasing process. Git will pause rebasing to allow you to sort
+out the conflicts. You do this the same way you solve merge conflicts,
+by checking all of the files git says have been changed in both histories
+and picking the versions you want. Be aware that these changes will show
+up in your pull request, so try and incorporate upstream changes as much
+as possible.
+
+You pick a file by `git add`ing it - you do not make commits during a
+rebase.
+
+Once you are done fixing conflicts for a specific commit, run:
+
+```bash
+git rebase --continue
+```
+
+This will continue the rebasing process. Once you are done fixing all
+conflicts you should run the existing tests to make sure you didn’t break
+anything, then run your new tests (there are new tests, right?) and
+make sure they work also.
+
+If rebasing broke anything, fix it, then repeat the above process until
+you get here again and nothing is broken and all the tests pass.
+
+### Make a pull request
+
+Make a clear pull request from your clone/fork and branch to the upstream develop
+branch, detailing exactly what changes you made and what feature this
+should add. The clearer your pull request is the faster you can get
+your changes incorporated into this repo.
+
+At least one other person MUST give your changes a code review, and once
+they are satisfied they will merge your changes into upstream. Alternatively,
+they may have some requested changes. You should make more commits to your
+branch to fix these, then follow this process again from rebasing onwards.
+
+Once you get back here, make a comment requesting further review and
+someone will look at your code again. If they like it, it will get merged,
+else, just repeat again.
+
+Thanks for contributing!
+
+### Guidelines
+
+1. Uphold the current code standard:
+    - Keep your code [DRY][].
+    - Apply the [boy scout rule][].
+    - Follow [STYLE-GUIDE.md](STYLE-GUIDE.md)
+1. Run the [tests][] before submitting a pull request.
+1. Tests are very, very important. Submit tests if your pull request contains
```

Comment:
I think it might be good to agree on a test coverage aim here as well. E.g: Code must have 80% or more coverage by testing suite to be mergeable.

Reply:
I think we should agree on a number and we can add it in.